stack/stack.rb

```diff
@@ -0,0 +1,56 @@
+  class Stack
+    attr_accessor :top
+    def initialize(size)
+      @size = size
+      @store = Array.new(@size)
+      @top = -1
+    end
+
+    def push(element)
+      if full?
+        raise "overflow error"
+      else
+        @top += 1
+        @store[@top] = element
+      end
+    end
+
+    def pop
+      if empty? or @store.nil?
+        raise "underflow error"
+      else
+        popped = @store[@top]
+        @store[@top] = nil
+        @top = @top.pred
+        popped
+      end
+    end
+
+    def remove_element(element)
+      if empty?
+        raise "underflow error"
+      else
+        deletion_index = nil
+        @store.each_with_index do |ele, idx|
+          if ele == element
+            deletion_index = idx
+            break
+          end
+        end
+        (@top - deletion_index + 1).times do |i|
```

Comment:
No need of this loop, inside the same loop also we can do changes.Can pop inside the above loop.